pkg/kn/flags/channel_types.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"knative.dev/client/pkg/kn/config"
+)
+
+type ChannelTypeFlags struct {
+	ctype string
+}
+
+func (i *ChannelTypeFlags) Add(f *pflag.FlagSet) {
+	f.StringVar(&i.ctype,
+		"type",
+		"",
+		"Override channel type to create, in the format '--type Group:Version:Kind'. "+
+			"If flag is not specified, it uses default messaging layer settings for channel type, cluster wide or specific namespace. "+
+			"You can configure aliases for channel types in kn config and refer the aliases with this flag. "+
+			"You can also refer inbuilt channel type InMemoryChannel using an alias 'imc' like '--type imc'. "+
+			"Examples: '--type messaging.knative.dev:v1alpha1:KafkaChannel' for specifying explicit Group:Version:Kind.")
+
+	for _, p := range config.GlobalConfig.ChannelTypeMappings() {
+		//user configration might override the default configuration
+		ctypeMappings[p.Alias] = schema.GroupVersionKind{
+			Kind:    p.Kind,
+			Group:   p.Group,
+			Version: p.Version,
+		}
+	}
+}
+
+// ctypeMappings maps aliases used for channel types to their GroupVersionKind
+var ctypeMappings = map[string]schema.GroupVersionKind{
+	"imcv1beta1": {
+		Group:   "messaging.knative.dev",
+		Version: "v1beta1",
+		Kind:    "InMemoryChannel",
+	},
+	"imcv1": {
```

Comment:
i see, colons are already used for gvk kind of specification. so i'm happy to use imcv1beta1 for the 'old' imc but please use 'imc' for the v1 as I expect this to be used much more often.

Reply:
@rhuss : fixed, PTAL.